src/petals/server/throughput.py

```diff
@@ -101,25 +101,25 @@ def measure_throughput_info(
     logger.info(
         "Measuring network and compute throughput. This takes about a minute and will be cached for future runs"
     )
-    return min(
-        measure_network_rps(config),
-        measure_compute_rps(
-            config, device, dtype, load_in_8bit=load_in_8bit, tensor_parallel_devices=tensor_parallel_devices
-        ),
-    )
-
 
-def measure_network_rps(config: BloomConfig) -> float:
+    result = measure_compute_rps(
+        config, device, dtype, load_in_8bit=load_in_8bit, tensor_parallel_devices=tensor_parallel_devices
+    )
     try:
-        s = speedtest.Speedtest()
-        s.get_servers()
-        s.get_best_server()
-        s.download()
-        s.upload()
-        network_info = s.results.dict()
-    except:
-        logger.error("Failed to measure network throughput:")
-        raise
+        result = min(result, measure_network_rps(config))
+    except Exception:
+        logger.warning("Failed to measure network throughput:", exc_info=True)
+        logger.warning("Proceeding with the compute throughput only")
+    return result
+
+
+def measure_network_rps(config: BloomConfig) -> Optional[float]:
+    s = speedtest.Speedtest()
+    s.get_servers()
+    s.get_best_server()
```

Comment:
here algorithm can be stuck for a long time

Reply:
You're right, this PR won't fix this. I'd suggest to still merge it, since it'll fix test flaps like this:
![Uploading Screenshot 2023-01-11 at 16.34.53.png…]()